app/src/test/java/com/example/parkingsystem/ParkingReservationPresenterTest.java

```diff
@@ -0,0 +1,331 @@
+        view = mock(FragmentReservationView.class);
+        presenter = new ReservationPresenter(model, view);
+    }
+
+    @Test
+    public void onReservationCreationButtonPressed_validData_reservationCreated() {
+        ReservationPresenter spyPresenter = spy(presenter);
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, 1);
+        Date startDate = calendar.getTime();
+        calendar.add(Calendar.DAY_OF_YEAR, 2);
+        Date endDate = calendar.getTime();
+        //Given
+        when(model.getParkingSize()).thenReturn(10);
+
+        when(view.getSecurityCode()).thenReturn("sd12");
+        when(view.getParkingLotNumberEntered()).thenReturn("2");
+        when(view.getStartDateTime()).thenReturn(startDate);
+        when(view.getEndDateTime()).thenReturn(endDate);
+        when(model.setParkingLotNumber("2")).thenReturn(2);
+
+        doReturn(true).when(spyPresenter).validateSecurityCode("A123");
+        doReturn(true).when(spyPresenter).validateParkingLotNumber(2);
+        doReturn(true).when(spyPresenter).validateDates(startDate.getTime(), endDate.getTime());
+
+        when(model.addReservationToParking(any(Reservation.class))).thenReturn(true);
+
+        //When
+        presenter.onReservationCreationButtonPressed();
+        //Then
+        verify(view).showReservationConfirmation();
+    }
+
+    @Test
+    public void onReservationCreationButtonPressed_anotherReservationInPlace_reservationNotAdded() {
+        ReservationPresenter spyPresenter = spy(presenter);
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, 1);
+        Date startDate = calendar.getTime();
+        calendar.add(Calendar.DAY_OF_YEAR, 2);
+        Date endDate = calendar.getTime();
+        //Given
+        when(model.getParkingSize()).thenReturn(10);
+
+        when(view.getSecurityCode()).thenReturn("sd12");
+        when(view.getParkingLotNumberEntered()).thenReturn("2");
+        when(view.getStartDateTime()).thenReturn(startDate);
+        when(view.getEndDateTime()).thenReturn(endDate);
+        when(model.setParkingLotNumber("2")).thenReturn(2);
+
+        doReturn(true).when(spyPresenter).validateSecurityCode("A123");
+        doReturn(true).when(spyPresenter).validateParkingLotNumber(2);
+        doReturn(true).when(spyPresenter).validateDates(startDate.getTime(), endDate.getTime());
+
+        when(model.addReservationToParking(any(Reservation.class))).thenReturn(false);
+
+        //When
+        presenter.onReservationCreationButtonPressed();
+        //Then
+        verify(view).showReservationNotAdded();
+    }
+
+    @Test
+    public void onReservationCreationButtonPressed_invalidLotNumber_catchException() {
+        //Given
+        when(model.getParkingSize()).thenReturn(2);
+        when(view.getParkingLotNumberEntered()).thenReturn("0");
+
+        doThrow(new IllegalArgumentException()).when(model).setParkingLotNumber("0");
```

Comment:
Can you do `when(model).setParkingLotNumber("0").thenThrow(IllegalArgumentException.class);`?

Reply:
No, if I change it the test breaks up.